lib/cog/models/relay.ex

```diff
@@ -24,11 +24,24 @@ defmodule Cog.Models.Relay do
     model
     |> Repo.preload(:groups)
     |> cast(params, @required_fields, @optional_fields)
+    |> allow_user_defined_id_on_insert(params)
     |> validate_presence_on_insert(:token)
     |> encode_token
     |> unique_constraint(:name)
   end
 
+  def allow_user_defined_id_on_insert(changeset, params) do
+    case {get_field(changeset, :id), Map.get(params, "id")} do
```

Comment:
Yeah, the reason I didn't include it in `@optional_fields` is that it's semantically different in that it's not actually optional if you're doing an update rather than a create. I'm not married to the way I did it, I was just trying to be more explicitly intention revealing.